### PR TITLE
Add kernel matrices as new features

### PR DESCRIPTION
**Pull request by qbarthelemy**
Solves https://github.com/pyRiemann/pyRiemann/issues/58

@plcrodrigues @gabelstein 

Accuracy of Laplacian kernels look promissing on this small example.
@sylvchev , can you test on MOABB if this improvement is statistically significant on large datasets?
Plot_cov_kernels

### CODE REVIEW

File: pyriemann/estimation.py [hunk 1]
```diff
@@ -704,6 +705,107 @@ def transform(self, X):
...
+        <https://www.cv-foundation.org/openaccess/content_iccv_2015/papers/Wang_Beyond_Covariance_Feature_ICCV_2015_paper.pdf>`_
+        L. Wang, J. Zhang, L. Zhou, C. Tang, W Li. ICCV, 2015.
+    .. [2]
+        https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.pairwise_kernels.html
+    """  # noqa
+
+    def __init__(self, metric='linear', n_jobs=None):
+        """Init."""
+        self.metric = metric
+        self.n_jobs = n_jobs
```
Comment (agramfort) [inline pyriemann/estimation.py:759]: some kernels have parameters eg the gamma in RBF. You should expose this option from the **kwds in the sklearn.metrics.pairwise.pairwise_kernels function

Comment (qbarthelemy) [reply pyriemann/estimation.py:759]: yep, good idea!

**Review by agramfort — APPROVED** [review]

**Review by sylvchev — APPROVED** [review]

**Comment by sylvchev** [comment]
Here are some results for larger datasets, on left/right hand motor imagery, with 5-fold CV on 6 datasets, averaging score accuracy for all subjects. I compared a classifier relying on covariance (baseline) with classifiers relying on kernels with different metrics (linear, polynomial, rbf, laplacian, cosine).

For MDM with covariance and MDM with various kernels
mdm-kernels

Same for tangent space projection + logistic regression
tslr-kernels

Finally for tangent space + SVM
tssvm-kernels

The improvement is visible for MDM with linear kernel, but no real improvement for tangent space with + classifier.

**Comment by sylvchev** [comment]
Results for MDM are nice and call for more research work!

LGTM!

**Comment by qbarthelemy** [comment]
Thx @sylvchev for this great validation!

For motor imagery classification, there is no significant differences with respect covariance matrices (equivalent to linear kernel, as explained in Wang2015).

Maybe, other kernels and/or other tasks will show accuracy improvement thanks to non-linear dependencies between temporal samples.
